library/std/src/sys/windows/compat.rs

```diff
@@ -103,20 +103,21 @@ macro_rules! compat_fn {
 
             #[allow(dead_code)]
             pub fn option() -> Option<F> {
-                unsafe { PTR }
+                unsafe {
+                    if cfg!(miri) {
+                        // Miri does not run `init`, so we just call `get_f` each time.
+                        get_f()
+                    } else {
+                        PTR
+                    }
+                }
             }
 
             #[allow(dead_code)]
             pub unsafe fn call($($argname: $argtype),*) -> $rettype {
-                if let Some(ptr) = PTR {
+                if let Some(ptr) = option() {
```

Comment:
This is super hot code, do we want `inline(always)` on `option` to ensure it doesn't get slowed down too much?